### PR TITLE
Upgrade configuration example and add toml example

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Necessary for #6527

### CODE REVIEW

File: examples/pylintrc [hunk 1]
```diff
@@ -1,5 +1,22 @@
 [MASTER]
 
+# Analyse import fallback blocks. This can be used to support both Python 2 and
+# 3 compatible code, which means that the block might have code that exists
+# only in one or another interpreter, leading to false positives when analysed.
+analyse-fallback-blocks=no
+
+# Load and enable all available extensions. Use --list-extensions to see a list
+# all available extensions.
+enable-all-extensions=None
```
Comment (Pierre-Sassoulas) [inline examples/pylintrc:10]: There are a lot of suspicious "None" value in this file that look like they should be empty strings.

File: examples/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+ignored-argument-names = "_.*|^ignored_|^unused_"
+
+# Tells whether we should check for unused import in __init__ files.
+# init-import =
+
+# List of qualified module names which can have objects that can redefine
+# builtins.
+redefining-builtins-modules = ["six.moves", "past.builtins", "future.builtins", "builtins", "io"]
+
+[tool.pylint."optional arguments"]
```
Comment (DanielNoord) [inline examples/pyproject.toml:527]: Hmm, this seems incorrect. I thought I fixed this...

Should probably look into this.

File: examples/pylintrc [hunk 3]
```diff
@@ -11,25 +28,39 @@ extension-pkg-allow-list=
...
+# List of module names for which member attributes should not be checked
+# (useful for modules/projects where namespaces are manipulated during runtime
+# and thus existing member attributes cannot be deduced by static analysis). It
+# supports qualified module names, as well as Unix pattern matching.
+ignored-modules=
 
 # Python code to execute, usually for sys.path manipulation such as
 # pygtk.require().
-#init-hook=
+init-hook=None
```
Comment (Pierre-Sassoulas) [inline examples/pylintrc:63]: Here in particular, I think we expect a string that will be executed. None is not wrong but it's misleading.

File: examples/pylintrc (hunk 1 above)
Comment (DanielNoord) [reply examples/pylintrc:10]: I was thinking the same. Is ``--generate-rcfile`` broken?

File: examples/pylintrc [hunk 4]
```diff
@@ -55,12 +93,45 @@ suggestion-mode=yes
 # active Python interpreter and may run arbitrary code.
 unsafe-load-any-extension=no
 
+# In verbose mode, extra non-checker-related info will be displayed.
+verbose=None
```
Comment (Pierre-Sassoulas) [inline examples/pylintrc:97]: Should be a boolean ?

File: examples/pylintrc (hunk 3 above)
Comment (DanielNoord) [reply examples/pylintrc:63]: I think ``--generate-rcfile`` might not handle the `pre-processed` options and their default values very well.

I handled those for ``--generate-toml-config`` but that might be broken for `rcfile`?

File: examples/pylintrc (hunk 1 above)
Comment (Pierre-Sassoulas) [reply examples/pylintrc:10]: ~~I don't think it's broken but~~, yes definitely broken:
````
Traceback (most recent call last):
  File "/home/pierre/pylint/venv/bin/pylint", line 11, in <module>
    load_entry_point('pylint', 'console_scripts', 'pylint')()
  File "/home/pierre/pylint/pylint/__init__.py", line 25, in run_pylint
    PylintRun(argv or sys.argv[1:])
  File "/home/pierre/pylint/pylint/lint/run.py", line 135, in __init__
    args = _config_initialization(
  File "/home/pierre/pylint/pylint/config/config_initialization.py", line 56, in _config_initialization
    linter._parse_configuration_file(config_args)
  File "/home/pierre/pylint/pylint/config/arguments_manager.py", line 242, in _parse_configuration_file
    self.config, parsed_args = self._arg_parser.parse_known_args(
  File "/usr/lib/python3.8/argparse.py", line 1812, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/lib/python3.8/argparse.py", line 2018, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/lib/python3.8/argparse.py", line 1958, in consume_optional
    take_action(action, args, option_string)
  File "/usr/lib/python3.8/argparse.py", line 1886, in take_action
    action(self, namespace, argument_values, option_string)
  File "/home/pierre/pylint/pylint/config/callback_actions.py", line 422, in __call__
    self.linter._load_reporters(values[0])
  File "/home/pierre/pylint/pylint/lint/pylinter.py", line 373, in _load_reporters
    reporter = self._load_reporter_by_name(reporter_name)
  File "/home/pierre/pylint/pylint/lint/pylinter.py", line 403, in _load_reporter_by_name
    raise exceptions.InvalidReporterError(name) from e
pylint.exceptions.InvalidReporterError: none
```

Comment (DanielNoord) [reply examples/pylintrc:10]: Guess we found another `beta-blocker` 😓

Comment (Pierre-Sassoulas) [reply examples/pylintrc:10]: I'm glad we thought about it a little and opened a new MR instead of rushing the release 😄

**Comment by Pierre-Sassoulas** [comment]
I don't think of anything else. Should we do the three at once or first fix then test ?

File: tbump.toml [hunk 5]
```diff
@@ -36,6 +36,18 @@ cmd = "tox -e docs||echo 'Hack so this command does not fail'"
 name = "Upgrade the pylintrc examples"
 cmd = "pylint --rcfile=/dev/null --generate-rcfile > examples/pylintrc"
 
+[[before_commit]]
```
Comment (DanielNoord) [inline tbump.toml:39]: After the merge of the other PRs these two steps are no longer necessary as the output of the commands is now checked in the normal test suite.

Comment (Pierre-Sassoulas) [reply tbump.toml:39]: [62cf71f](https://github.com/PyCQA/pylint/pull/6529/commits/62cf71f65a435dfdb556a6650d08ece7469d861f)

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**